garden/migration.md

```diff
@@ -0,0 +1,465 @@
+
+</details>
+
+## Untocks
+
+A very small selection of things have not been migrated, mostly for backwards compatibility reasons (e.g. supporting Fortress.)
+
+- Branch names for versions of Gazebo libraries targeting releases before Garden
+    - e.g. `ign-cmake1`
+- Some links
+    - e.g. [https://osrf-migration.github.io/ignition-gh-pages](https://osrf-migration.github.io/ignition-gh-pages)
+- Fuel user agent related
+    - e.g. `X-Ign-Resource-Version`, `IgnitionFuelTools`
+
+# Migration
+
+## Overview
+
+The following migration guidelines are just that—guidelines and suggestions for how to get your package migrated, that hopefully should cover most general cases.
+
+Just keep in mind the overarching goal of replacing every Ignition counterpart (`IGN`, `Ign`, `Ignition`, `ign`, `ignition`) to the Gazebo counterpart (`GZ`, `Gz`, `gz`).
+
+### Recommendations
+
+What might help greatly is to:
+
+- Make ample use of regex and `sed`
+    - The migration guide will be giving suggestions for what regex expressions to use, but **always review before executing the replace**
+- Pay attention to case!!
+- Use an editor to review changes, and source control to allow for easy rollbacks
+- Pay attention to compilation warnings/errors, which usually (but not always) give suggestions for what to replace!
+- If in doubt, trace the change to changes listed in the [tracking issue](https://github.com/gazebo-tooling/release-tools/issues/698)
+
+Also, if you are building the Gazebo stack from source, you should do a clean, fresh rebuild and install. Delete your `build` and `install` directories, and run the build with `--merge-install`.
+
+### Gotchas
+
+Most of the migration effort you should do is mainly finding and replacing instances of Ignition-related terms with the Gazebo-related one.
+
+However, here are a lot of edge cases that make it difficult to create a script to handle the migration (leading to errors or bugs), so before the migration steps are suggested, it would be good to know some of these cases.
+
+- Case-sensitivity not respected
+    - e.g. `IGNITION_ADD_PLUGIN` → `gz_ADD_PLUGIN`
+- Greedy matches
+    - e.g. `Align` → `Algz`, `unsigned` → `unsgzed`, `signal` → `sgzal`
+- Not migrating “gazebo” to “sim”
+    - e.g. `ign-gazebo` → `gz-gazebo` (it should be `gz-sim`)
+- Matching `ign` before `ignition`
+    - e.g. `ignition-cmake3` → `gzition-cmake3`
+- Migrating a library that should not be migrated
+    - e.g. `ignition-cmake2` → `gz-cmake2`
+- Grammar
+    - e.g. “an Ignition library” → “**an** Gazebo library”
+
+Additionally, take note of the following behaviors:
+
+- Config files are not overwritten, so you might need to delete or manually migrate Ignition-counterpart references in your config files to get them to point to the appropriate Gazebo-counterpart locations
```

Comment:
It would be good to add an example here. I'm not sure this applies to all config files.

Reply:
[`bbe100c` (#275)](https://github.com/gazebosim/docs/pull/275/commits/bbe100c59fd81652c1325a27519af1fe126c4fca)

Reply:
Not sure about this example:

> For example, a `sim/7/server.config` file could be using an `ignition::gazebo::systems::Physics` plugin.

This is the config file specific to Garden, and we install it, so it should be correct from day 1.

Reply:
Ping @methylDragon

Reply:
Refined the example, hopefully this works :crossed_fingers: 

https://github.com/gazebosim/docs/pull/275/commits/58f6ec9d15ee7d8ae6fedcdcabc1e01baba67f26